src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -0,0 +1,136 @@
+      SwerveModuleConstants... modules) {
+
+    // Sets the drivetrain constants, odometry update frequency and constants for
+    // the swerve modules
+    super(driveTrainConstants, OdometryUpdateFrequency, modules);
+
+    // Sets the supply current limits for the swerve modules (for driving and
+    // turning)
+    CurrentLimitsConfigs driveCurrentLimits =
+        new CurrentLimitsConfigs()
+            .withSupplyCurrentLimit(Constants.Swerve.kDriveSupplyCurrentLimit)
+            .withSupplyCurrentLimitEnable(true);
+
+    CurrentLimitsConfigs turningCurrentLimits =
+        new CurrentLimitsConfigs()
+            .withSupplyCurrentLimit(Constants.Swerve.kTurningSupplyCurrentLimit)
+            .withSupplyCurrentLimitEnable(true);
+
+    for (SwerveModule module : Modules) {
+      module.getDriveMotor().getConfigurator().apply(driveCurrentLimits);
+      module.getSteerMotor().getConfigurator().apply(turningCurrentLimits);
+    }
+
+    // Configures the holonomic auto builder
+    configurePathPlanner();
+  }
+
+  // Constructor for default odometry update frequency
+  public SwerveSubsystem(
+      SwerveDrivetrainConstants driveTrainConstants, SwerveModuleConstants... modules) {
+    this(driveTrainConstants, 0, modules);
+  }
+
+  public static SwerveSubsystem getInstance() {
+    if (instance == null) {
+      instance =
+          new SwerveSubsystem(
+              Constants.Swerve.DrivetrainConstants,
+              Constants.Swerve.FrontLeft,
+              Constants.Swerve.FrontRight,
+              Constants.Swerve.BackLeft,
+              Constants.Swerve.BackRight);
+    }
+    return instance;
+  }
+
+  // Applies swerve request to drivetrain
+  public Command applyRequest(Supplier<SwerveRequest> requestSupplier) {
+    return run(() -> this.setControl(requestSupplier.get()));
+  }
+
+  private final SwerveRequest.ApplyChassisSpeeds autoRequest =
+      new SwerveRequest.ApplyChassisSpeeds();
+
+  private void configurePathPlanner() {
+    // Sets radius of drive base (using module locations)
+    double driveBaseRadius = 0;
+    for (var moduleLocation : m_moduleLocations) {
+      driveBaseRadius = Math.max(driveBaseRadius, moduleLocation.getNorm());
+    }
+
+    // Holonomic auto builder
+    AutoBuilder.configureHolonomic(
+        () -> this.getState().Pose, // Supplier of current robot pose
+        this::seedFieldRelative, // Consumer for seeding pose against auto
+        this::getCurrentRobotChassisSpeeds,
+        (speeds) ->
+            this.setControl(
+                autoRequest.withSpeeds(speeds)), // Consumer of ChassisSpeeds to drive the
+        // robot
+        new HolonomicPathFollowerConfig(
+            new PIDConstants(3, 0, 0), // CHANGE FOR NEW ROBOT
+            new PIDConstants(3, 0, 0),
+            Constants.Swerve.kSpeedAt12VoltsMps,
+            driveBaseRadius,
+            new ReplanningConfig()),
+        () -> false, // Change this if the path needs to be flipped on red vs blue
```

Comment:
I think there's a bit more logic to this than just false, because it needs to be true for any pathplanner paths because the field is mirrored, but may not be flipped for aligning using pose estimation.